rasa/graph_components/providers/nlu_training_data_provider.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+from typing import Dict, Text, Any, Optional
+import os
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.importers.importer import TrainingDataImporter
+from rasa.shared.nlu.training_data.training_data import (
+    TrainingData,
+    DEFAULT_TRAINING_DATA_OUTPUT_PATH,
+)
+from rasa.shared.nlu.training_data.loading import load_data
+
+
+class NLUTrainingDataProvider(GraphComponent):
+    """Provides NLU training data during training and inference time."""
+
+    def __init__(
+        self,
+        model_storage: ModelStorage,
+        resource: Resource,
+        training_data: Optional[TrainingData] = None,
+    ) -> None:
+        """Creates NLU training data provider."""
+        self._model_storage = model_storage
+        self._resource = resource
+        self._training_data = training_data
+
+    @classmethod
+    def get_default_config(cls) -> Dict[Text, Any]:
+        return {"language": "en", "persist": False}
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> NLUTrainingDataProvider:
+        """Creates component (see parent class for full docstring)."""
+        return cls(model_storage, resource)
```

Comment:
Is it okay that we have unused arguments in some of the class methods (create and load)? Do we plan to extend the class and use them? Or are they there to maintaini a consistent interface regardless?

Reply:
> are they there to maintaini a consistent interface regardless?

Yep, exactly :) This makes it easy to create every graph component in exactly the same way. If a component isn't configurable or doesn't need to persist anything (because nothing needs to be "trained"), then they can just leave those parameters unused (e.g. the `graph_component/converter/nlu_message_converter.py` doesn't need anything).

(Although I think in the future this will change a little - at least the "config" should be always passed on to enable checks - i.e. if the component can't be configured, we shouldn't be able to attempt to configure it without raising a warning)

Reply:
... so the config needs to be passed on to the constructor here!